### PR TITLE
[cdc-connector][oracle] Expose SNAPSHOT_ONLY mode for Oracle CDC Source

### PR DESCRIPTION
**Pull request by loserwang1024**
As shown in https://github.com/ververica/flink-cdc-connectors/issues/2867, expose SNAPSHOT_ONLY mode for Oracle CDC Source  after until https://github.com/ververica/flink-cdc-connectors/issues/2909 is fixed.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@GOODBOY008 , @leonardBang , CC

**Comment by loserwang1024** [comment]
@GOODBOY008 , CC

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oracle-cdc/src/main/java/com/ververica/cdc/connectors/oracle/source/reader/fetch/EventProcessorFactory.java [hunk 1]
```diff
@@ -240,12 +242,13 @@ public CDCRemoteInfinispanLogMinerEventProcessor(
...
-            super.processRow(partition, row);
-            afterRowProcess(partition, row, redoLogSplit, errorHandler, dispatcher, context);
+            if (beforeRowProcess(partition, row, redoLogSplit, errorHandler, dispatcher, context)) {
+                super.processRow(partition, row);
+            }
         }
     }
 
-    public static void afterRowProcess(
+    public static boolean beforeRowProcess(
```
Comment (GOODBOY008) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-oracle-cdc/src/main/java/com/ververica/cdc/connectors/oracle/source/reader/fetch/EventProcessorFactory.java:251]: I think mix the logic in this method with this name , maybe case confuse.

**Review by GOODBOY008 — COMMENTED** [review]
@loserwang1024  Thanks for your contribution.General is good, but I left minor comment.

**Review by GOODBOY008 — APPROVED** [review]
LGTM

**Pull request merged**